packages/contracts/test/diamond/facets/StakingFacet.t.sol

```diff
@@ -1,376 +1,630 @@
+        assertEq(governancePerBlock, 3);
+        assertEq(governanceTreasuryDivider, 4);
+        assertEq(rewardAmount, 10 ether);
+        assertEq(totalAllocationPoints, 100);
+        assertEq(startBlock, 20);
+    }
+
+    function testGetStakingUserInfo_ShouldReturnStakingUserInfo() public {
+        // user stakes 50 STK
+        vm.prank(user);
+        stakingFacet.stake(0, 50 ether);
+
+        // 10 blocks pass
+        vm.roll(block.number + 10);
+
+        // user stakes 50 STK
+        vm.prank(user);
+        stakingFacet.stake(0, 50 ether);
+
+        LibStaking.UserInfo memory userInfo = stakingFacet.getStakingUserInfo(
+            0,
+            user
         );
+        assertEq(userInfo.amount, 100 ether);
+        assertEq(userInfo.rewardDebt, 20 ether);
+    }
 
-        accessControlFacet.grantRole(
-            GOVERNANCE_TOKEN_BURNER_ROLE,
-            address(diamond)
+    function testGetStakingPoolInfo_ShouldReturnStakingPoolInfo() public {
+        // user stakes 50 STK
+        vm.prank(user);
+        stakingFacet.stake(0, 50 ether);
+
+        // 10 blocks pass
+        vm.roll(block.number + 10);
+
+        // refresh pool
+        stakingFacet.updateStakingPool(0);
+
+        LibStaking.PoolInfo memory poolInfo = stakingFacet.getStakingPoolInfo(
+            0
         );
-        managerFacet.setCreditTokenAddress(address(creditToken));
+        assertEq(address(poolInfo.lpToken), address(stakeToken));
+        assertEq(poolInfo.amount, 50 ether);
+        assertEq(poolInfo.allocationPoints, 100);
+        assertEq(poolInfo.lastRewardBlock, 11);
+        assertEq(poolInfo.accumulatedGovernancePerShare, 0.0000002 ether);
+    }
 
-        vm.stopPrank();
+    function testGetStakingPoolsLength_ShouldReturnNumberOfStakingPools()
+        public
+    {
+        uint256 poolsLength = stakingFacet.getStakingPoolsLength();
+        assertEq(poolsLength, 1);
+    }
 
-        vm.startPrank(stakingMinAccount);
-        dollarToken.approve(address(metapool), 10000e18);
-        crvToken.approve(address(metapool), 10000e18);
-        vm.stopPrank();
+    //==================
+    // Public methods
+    //==================
 
-        vm.startPrank(stakingMaxAccount);
-        dollarToken.approve(address(metapool), 10000e18);
-        crvToken.approve(address(metapool), 10000e18);
-        vm.stopPrank();
-        vm.startPrank(fourthAccount);
-        dollarToken.approve(address(metapool), 10000e18);
-        crvToken.approve(address(metapool), 10000e18);
-        vm.stopPrank();
+    function testMassUpdateStakingPools_ShouldUpdateAllStakingPools() public {
+        // before
+        LibStaking.PoolInfo memory poolInfo = stakingFacet.getStakingPoolInfo(
+            0
+        );
+        assertEq(poolInfo.lastRewardBlock, 1);
+
+        // 10 blocks pass
+        vm.roll(block.number + 10);
+
+        stakingFacet.massUpdateStakingPools();
+
+        // after
+        poolInfo = stakingFacet.getStakingPoolInfo(0);
```

Comment:
we need to test this func with more pools if we want it to be massive

Reply:
Added in https://github.com/ubiquity/ubiquity-dollar/pull/987/commits/c57b8f0ad9e2f380cdb9d7378f54424f3cd2e0a9